features/step_definitions/networking.rb

```diff
@@ -1031,70 +1031,81 @@
-  # Leader: self
-  if leader_id.nil? || leader_id[1] == "unknown"
-    raise "Unknown leader"
-  end
-  servers = cluster_state.match(/Servers:\n(.*)/m)
-  leader_line = servers[1].lines.find { |line| line.include? "(" + leader_id[1] }
-  unless leader_line
-    raise "Unable to find leader #{leader_id[1]}"
-  end
-  # Servers:
-  #   6e24 (6e24 at ssl:[fd2e:6f44:5dd8::81]:9643) (self) next_index=11214 match_index=11517
-  #   90fb (90fb at ssl:[fd2e:6f44:5dd8::68]:9643) next_index=11518 match_index=11517 last msg 350 ms ago
-  # Servers:
-  #   c977 (c977 at ssl:172.31.248.170:9643) next_index=3573 match_index=3572
-  #   d73f (d73f at ssl:172.31.248.168:9643) (self) next_index=3265 match_index=3572
-  # match first string in the parens, the everything from the first colon to a colon digit close-paren sequence
-  splits = leader_line.match(/\((\S+)[^:]+:\[?([^\]\[)]+)\]?:(\d+)\)/)
-  leader_node = splits.captures[1]
-  leader_pod = BushSlicer::Pod.get_labeled("app=ovnkube-master", project: project("openshift-ovn-kubernetes", switch: false),
-                                           user: admin, quiet: true) { |pod, hash|
-    pod.node_name == leader_node || pod.ip == leader_node
-  }.first
-  # update the cache so we can execute on the pod without specify the name
-  cache_resources leader_pod
-  cb[cb_leader_name] = leader_pod
-  logger.info "cb.#{cb_leader_name}.name = #{leader_pod.name}"
-  logger.info "cb.#{cb_leader_name}.node_name = #{leader_pod.node_name}"
+    servers = cluster_state.match(/Servers:\n(.*)/m)
+    leader_line = servers[1].lines.find { |line| line.include? "(" + leader_id[1] }
+    unless leader_line
+      raise "Unable to find leader #{leader_id[1]}"
+    end
+    # Servers:
+    #   6e24 (6e24 at ssl:[fd2e:6f44:5dd8::81]:9643) (self) next_index=11214 match_index=11517
+    #   90fb (90fb at ssl:[fd2e:6f44:5dd8::68]:9643) next_index=11518 match_index=11517 last msg 350 ms ago
+    # Servers:
+    #   c977 (c977 at ssl:172.31.248.170:9643) next_index=3573 match_index=3572
+    #   d73f (d73f at ssl:172.31.248.168:9643) (self) next_index=3265 match_index=3572
+    # match first string in the parens, the everything from the first colon to a colon digit close-paren sequence
+    splits = leader_line.match(/\((\S+)[^:]+:\[?([^\]\[)]+)\]?:(\d+)\)/)
+    leader_node = splits.captures[1]
+    leader_pod = BushSlicer::Pod.get_labeled("app=ovnkube-master", project: project("openshift-ovn-kubernetes", switch: false),
+                                             user: admin, quiet: true) { |pod, hash|
+      pod.node_name == leader_node || pod.ip == leader_node
+    }.first
+    # update the cache so we can execute on the pod without specify the name
+    cache_resources leader_pod
+    cb[cb_leader_name] = leader_pod
+    logger.info "cb.#{cb_leader_name}.name = #{leader_pod.name}"
+    logger.info "cb.#{cb_leader_name}.node_name = #{leader_pod.node_name}"
+  else  
+    step "I select a random node's host"
```

Comment:
@anuragthehatter  QQ for random node?  I have not check the OVN_IC details yet? just know each worker will have own DB. see if podA is created on workerA,  However here select a random node(workerB), it that mean the podA info can be also find in workerB DB?

Reply:
Yea so this is still being evaluated as per I commented on https://github.com/openshift/openshift-tests-private/pull/10506#discussion_r1263817853

But for now seems like exisiting usecsaes with this change should work. I will be making more changes as i test further. Thanks

Reply:
SUbmitted my all fixes. Please check your cases @openshift/team-sdn-qe . Changes seem to be working fine on both 4.13 and 4.14 (pre-IC bot cluster)